client/admin.lua

```diff
@@ -293,229 +293,222 @@ end
 
 local ShowBlips = false
 local ShowNames = false
-local NetCheck1 = false
-local NetCheck2 = false
-
-CreateThread(function()
-    while true do
-        Wait(1000)
-        if NetCheck1 or NetCheck2 then
-            TriggerServerEvent('qb-admin:server:GetPlayersForBlips')
-        end
-    end
-end)
+local BlipCheck = false
+local NameCheck = false
 
 RegisterNetEvent('qb-admin:client:blips', function()
-    if not ShowBlips then
-        ShowBlips = true
-        NetCheck1 = true
-        lib.notify({ description = Lang:t('success.blips_activated'), type = 'success' })
-    else
-        ShowBlips = false
-        lib.notify({ description = Lang:t('error.blips_deactivated'), type = 'error' })
-    end
+    ShowBlips = not ShowBlips
+    BlipCheck = ShowBlips or BlipCheck
+    ShowNamesBlips()
```

Comment:
this function should only be called if ShowBlips is true, otherwise it creates a race condition where threadRunning may be false by the time this function executes, causing the blips to stay on, even when the event calls to turn them off.